frontend/src/features/permits/pages/Payment/PaymentRedirect.tsx

```diff
@@ -97,24 +56,34 @@ export const PaymentRedirect = () => {
   }, [paymentDetails.trnApproved]);
 
   useEffect(() => {
-    if (issuedPermit.current === false) {
-      const permitIdsArray = getPermitIdsArray(permitIds);
 
-      if (permitIdsArray.length === 0) {
-        // permit ids should not be empty, if so then something went wrong
-        navigate(ERROR_ROUTES.UNEXPECTED, { replace: true });
-      } else if (paymentApproved === true) {
+    const applicationIds:string[] = [];
+    if (transactionIdQuery?.isSuccess && issuedPermit?.current === false) {
+      
+      transactionIdQuery?.data?.applicationDetails?.forEach((application) => {
+        if (application?.applicationId) {
+          applicationIds.push(application.applicationId);
+        }
+      })
+
+      if (paymentApproved === true) {
         // Payment successful, proceed to issue permit
-        issuePermitsMutation.mutate(permitIdsArray);
+        issuePermitsMutation.mutate(applicationIds);
         issuedPermit.current = true;
       } else if (paymentApproved === false) {
         // Payment failed, redirect back to pay now page
-        navigate(APPLICATIONS_ROUTES.PAY(permitIdsArray[0], true), {
+        navigate(APPLICATIONS_ROUTES.PAY(applicationIds[0], true), {
           replace: true,
         });
       }
     }
-  }, [paymentApproved, permitIds]);
+
+    // if the query status has completed (aka. "idle") and the query failed, navigate
+    // to error page
+    if (transactionIdQuery?.fetchStatus === 'idle' && !transactionIdQuery?.isSuccess) {
```

Comment:
Does `!transactionIdQuery?.isSuccess` work at all? Your getPaymentByTransactionId function masquerades any errors and returns null. I think it will always be a success scenario regardless of API sending non-200. Can you confirm?

Reply:
yeah it "worked" but not accurately trapping the failing condition. Reworked the API call to allow us more visibility into the success or failure of that query.